script/GameObjects/Components/Object/Yield.lua

```diff
@@ -4,34 +4,100 @@ local Entity = require('GameObjects.Entity')
 --        than 'size' (energy-normalized) units of item may be extracted per
 --        unit time from the Entity.
 local Yield = class(function (self, item, size)
-  self.item = item
-  self.size = size
+  self.item     = item
+  self.size     = size
+  self.maxSize  = size
+  self.cooldown = 300 -- ore yields start to respawn after 5 minutes after last mining
+  self.duration =   5 -- ore yields respawn one unit every 5 seconds
+  self.ctimer   = 0.0
```

Comment:
Could be clearer that these are the cooldown timer and "duration" timer?

Reply:
Josh used "etimer" in MineAt.lua; I followed that convention here in Yield.lua. That said, I've updated this section to provide detailed comments on how these variables are used; those changes will show up as soon as I do a new commit following my review of the other suggestions made.